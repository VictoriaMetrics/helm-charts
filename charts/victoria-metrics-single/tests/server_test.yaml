suite: server templates test
chart:
  version: 0.1.1
  appVersion: 0.1.0
templates:
  - server.yaml
tests:
  - it: statefulSet should match snapshot
    asserts:
      - matchSnapshot: {}
    set: {}
  - it: statefulSet with volume should match snapshot
    asserts:
      - matchSnapshot: {}
    set:
      server:
        persistentVolume:
          enabled: true
  - it: deployment should match snapshot
    set:
      server:
        mode: deployment
    asserts:
      - matchSnapshot: {}
  - it: deployment with volume should match snapshot
    set:
      server:
        mode: statefulSet
        persistentVolume:
          enabled: true
    asserts:
      - matchSnapshot: {}
