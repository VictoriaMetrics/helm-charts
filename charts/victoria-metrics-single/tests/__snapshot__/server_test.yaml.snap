deployment should match snapshot:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: server
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: victoria-metrics-single
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: victoria-metrics-single-0.1.1
      name: RELEASE-NAME-victoria-metrics-single-server
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app: server
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: victoria-metrics-single
      strategy:
        type: Recreate
      template:
        metadata:
          labels:
            app: server
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: victoria-metrics-single
            app.kubernetes.io/version: 0.1.0
            helm.sh/chart: victoria-metrics-single-0.1.1
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - --envflag.enable
                - --envflag.prefix=VM_
                - --httpListenAddr=:8428
                - --loggerFormat=json
                - --retentionPeriod=1
                - --storageDataPath=/storage
              image: victoriametrics/victoria-metrics:0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 10
                initialDelaySeconds: 30
                periodSeconds: 30
                tcpSocket:
                  port: http
                timeoutSeconds: 5
              name: vmsingle
              ports:
                - containerPort: 8428
                  name: http
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /health
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 5
                periodSeconds: 15
                timeoutSeconds: 5
              securityContext: {}
              volumeMounts:
                - mountPath: /storage
                  name: server-volume
          securityContext: {}
          serviceAccountName: RELEASE-NAME-victoria-metrics-single-server
          terminationGracePeriodSeconds: 60
          volumes:
            - name: server-volume
              persistentVolumeClaim:
                claimName: RELEASE-NAME-victoria-metrics-single-server
deployment with volume should match snapshot:
  1: |
    apiVersion: apps/v1
    kind: StatefulSet
    metadata:
      labels:
        app: server
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: victoria-metrics-single
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: victoria-metrics-single-0.1.1
      name: RELEASE-NAME-victoria-metrics-single-server
      namespace: NAMESPACE
    spec:
      podManagementPolicy: OrderedReady
      replicas: 1
      selector:
        matchLabels:
          app: server
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: victoria-metrics-single
      serviceName: RELEASE-NAME-victoria-metrics-single-server
      template:
        metadata:
          labels:
            app: server
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: victoria-metrics-single
            app.kubernetes.io/version: 0.1.0
            helm.sh/chart: victoria-metrics-single-0.1.1
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - --envflag.enable
                - --envflag.prefix=VM_
                - --httpListenAddr=:8428
                - --loggerFormat=json
                - --retentionPeriod=1
                - --storageDataPath=/storage
              image: victoriametrics/victoria-metrics:0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 10
                initialDelaySeconds: 30
                periodSeconds: 30
                tcpSocket:
                  port: http
                timeoutSeconds: 5
              name: vmsingle
              ports:
                - containerPort: 8428
                  name: http
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /health
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 5
                periodSeconds: 15
                timeoutSeconds: 5
              securityContext: {}
              volumeMounts:
                - mountPath: /storage
                  name: server-volume
          securityContext: {}
          serviceAccountName: RELEASE-NAME-victoria-metrics-single-server
          terminationGracePeriodSeconds: 60
          volumes: null
      updateStrategy: {}
      volumeClaimTemplates:
        - apiVersion: v1
          kind: PersistentVolumeClaim
          metadata:
            name: server-volume
          spec:
            accessModes:
              - ReadWriteOnce
            resources:
              requests:
                storage: 16Gi
statefulSet should match snapshot:
  1: |
    apiVersion: apps/v1
    kind: StatefulSet
    metadata:
      labels:
        app: server
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: victoria-metrics-single
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: victoria-metrics-single-0.1.1
      name: RELEASE-NAME-victoria-metrics-single-server
      namespace: NAMESPACE
    spec:
      podManagementPolicy: OrderedReady
      replicas: 1
      selector:
        matchLabels:
          app: server
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: victoria-metrics-single
      serviceName: RELEASE-NAME-victoria-metrics-single-server
      template:
        metadata:
          labels:
            app: server
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: victoria-metrics-single
            app.kubernetes.io/version: 0.1.0
            helm.sh/chart: victoria-metrics-single-0.1.1
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - --envflag.enable
                - --envflag.prefix=VM_
                - --httpListenAddr=:8428
                - --loggerFormat=json
                - --retentionPeriod=1
                - --storageDataPath=/storage
              image: victoriametrics/victoria-metrics:0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 10
                initialDelaySeconds: 30
                periodSeconds: 30
                tcpSocket:
                  port: http
                timeoutSeconds: 5
              name: vmsingle
              ports:
                - containerPort: 8428
                  name: http
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /health
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 5
                periodSeconds: 15
                timeoutSeconds: 5
              securityContext: {}
              volumeMounts:
                - mountPath: /storage
                  name: server-volume
          securityContext: {}
          serviceAccountName: RELEASE-NAME-victoria-metrics-single-server
          terminationGracePeriodSeconds: 60
          volumes: null
      updateStrategy: {}
      volumeClaimTemplates:
        - apiVersion: v1
          kind: PersistentVolumeClaim
          metadata:
            name: server-volume
          spec:
            accessModes:
              - ReadWriteOnce
            resources:
              requests:
                storage: 16Gi
statefulSet with volume should match snapshot:
  1: |
    apiVersion: apps/v1
    kind: StatefulSet
    metadata:
      labels:
        app: server
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: victoria-metrics-single
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: victoria-metrics-single-0.1.1
      name: RELEASE-NAME-victoria-metrics-single-server
      namespace: NAMESPACE
    spec:
      podManagementPolicy: OrderedReady
      replicas: 1
      selector:
        matchLabels:
          app: server
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: victoria-metrics-single
      serviceName: RELEASE-NAME-victoria-metrics-single-server
      template:
        metadata:
          labels:
            app: server
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: victoria-metrics-single
            app.kubernetes.io/version: 0.1.0
            helm.sh/chart: victoria-metrics-single-0.1.1
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - --envflag.enable
                - --envflag.prefix=VM_
                - --httpListenAddr=:8428
                - --loggerFormat=json
                - --retentionPeriod=1
                - --storageDataPath=/storage
              image: victoriametrics/victoria-metrics:0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 10
                initialDelaySeconds: 30
                periodSeconds: 30
                tcpSocket:
                  port: http
                timeoutSeconds: 5
              name: vmsingle
              ports:
                - containerPort: 8428
                  name: http
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /health
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 5
                periodSeconds: 15
                timeoutSeconds: 5
              securityContext: {}
              volumeMounts:
                - mountPath: /storage
                  name: server-volume
          securityContext: {}
          serviceAccountName: RELEASE-NAME-victoria-metrics-single-server
          terminationGracePeriodSeconds: 60
          volumes: null
      updateStrategy: {}
      volumeClaimTemplates:
        - apiVersion: v1
          kind: PersistentVolumeClaim
          metadata:
            name: server-volume
          spec:
            accessModes:
              - ReadWriteOnce
            resources:
              requests:
                storage: 16Gi
